src/envoy/auth/config.cc

```
@@ -15,199 +15,119 @@
+    }
   }
-}
-void AsyncClientCallbacks::onFailure(AsyncClient::FailureReason) {
-  ENVOY_LOG(debug, "AsyncClientCallbacks [cluster = {}]: failed",
-            cluster_->name());
-  cb_(false, "");
+  return "";
 }
 
-void AsyncClientCallbacks::Call(const std::string &uri) {
-  ENVOY_LOG(debug, "AsyncClientCallbacks [cluster = {}]: {} {}",
-            cluster_->name(), __func__, uri);
-  // Example:
-  // uri  = "https://example.com/certs"
-  // pos  :          ^
-  // pos1 :                     ^
-  // host = "example.com"
-  // path = "/certs"
-  auto pos = uri.find("://");
-  pos = pos == std::string::npos ? 0 : pos + 3;  // Start position of host
-  auto pos1 = uri.find("/", pos);
-  if (pos1 == std::string::npos) pos1 = uri.length();
-  std::string host = uri.substr(pos, pos1 - pos);
-  std::string path = "/" + uri.substr(pos1 + 1);
-
-  MessagePtr message(new RequestMessageImpl());
-  message->headers().insertMethod().value().setReference(
-      Http::Headers::get().MethodValues.Get);
-  message->headers().insertPath().value(path);
-  message->headers().insertHost().value(host);
-
-  request_ = cm_.httpAsyncClientForCluster(cluster_->name())
-                 .send(std::move(message), *this, timeout_);
+Config::Config(std::vector<IssuerInfo>&& issuers)
+    : issuers_(std::move(issuers)) {
+  ENVOY_LOG(debug, "Config: {}", __func__);
 }
 
-void AsyncClientCallbacks::Cancel() { request_->cancel(); }
+Config::Config(const Json::Object& config) {
+  ENVOY_LOG(debug, "Config: {}", __func__);
 
-IssuerInfo::IssuerInfo(Json::Object *json) {
-  ENVOY_LOG(debug, "IssuerInfo: {}", __func__);
-  // Check "name"
-  name_ = json->getString("name", "");
-  if (name_ == "") {
-    ENVOY_LOG(debug, "IssuerInfo: Issuer name missing");
-    failed_ = true;
+  // Load the issuers as JSON array.
+  std::vector<Json::ObjectSharedPtr> issuer_jsons;
+  try {
+    issuer_jsons = config.getObjectArray("issuers");
+  } catch (...) {
```

Do not do `try` and `catch` here and below, the exception thrown by Envoy `getObjectArray` is clear enough and it will be logged with stacktrace, which is more clear, and simply return below will swallow the error.